lib/transaction/pool.go

```diff
@@ -57,22 +66,26 @@ func (tp *Pool) GetFromSource(source string) (Transaction, bool) {
 	return tp.Get(hash)
 }
 
-func (tp *Pool) Add(tx Transaction) bool {
+func (tp *Pool) Add(tx Transaction) error {
 	txHash := tx.GetHash()
 	if tp.Has(txHash) {
-		return false
+		return errors.TransactionAlreadyExistsInPool
 	}
 
 	tp.Lock()
 	defer tp.Unlock()
 
+	if len(tp.Pool) >= tp.limit {
```

Comment:
If we use lock strictly, the upper locking part should be moved under this part.

Reply:
The check `len(tp.Pool)` in the lock is more strictly. Is it not right?